### ISSUE TITLE
[REQ] As a user I want to have the entries sorted

### ISSUE THREAD

**Issue by tangrit** [jobe-m/ownkeepass#82]
Having a lot of groups in top level or another group it is with an effort to find the right one as groups are not searchable. But even for the entries itself would it be more clearly if they are neatly arranged.

Sorting key should be changeable: title, user, inet-address, expiration Date (for later implementation)
Most important is the title of course :)
BTW: it would be nice to have a field "email address" since often internet accounts are associated with an email address.

**Comment by jobe-m** [comment]
BTW custom entries like email address cannot be added easily with the Keepass 1 database format. This will be doable when using Keepass 2 database format. There it is possible to specify new items within a password entry.

Otherwise sorting would be nice of course. I acknowledge that.
Sorting also depends on the capabilities of the database otherwise after a restart of the app the sorting would be lost. But it is definitely possible to sort in custom order or alphabetically with Keepass 1 database format. So contributions for that are highly appreciated :)

**Comment by jobe-m** [comment]
Details for implementation:
* An entry can be moved inside a group by writing its new position to the "entry function" setVisualIndex(int index) in the database interface (backend). This is used to sort the password entries in custom order. Check before if entry is valid with isValid() function.
* To sort the password entries and groups within a group alphabetically in ascending order (title, username) use the following two database interface functions to get a sorted list of entries and groups: QList<IEntryHandle*> entriesSortedStd(IGroupHandle* Group), QList<IGroupHandle*> sortedGroups()

**Comment by tangrit** [comment]
I would be glad if I had something more to contribute than ideas.
For myself I started coding with Qt 1.44 and my last experience was porting programs from v2 to v3 some time ago. I did no understand why the sort functions already implemented in ownkeepass 1.0.24 take no effect. However, Qt-doc (http://doc.qt.io/qt-5/qtalgorithms.html) suggests to use  the STL equivalents (std::sort).

May be we can do talk in german aside.

**Comment by jobe-m** [comment]
No problem. This was no request to anybody ;)
I just had some time between two appointments and looked into the code and wrote some ideas down. In the appropriate place I hope.
BTW There is no sorting implemented as of now. The order of password entries in a group is that of its creation. The latest is always at the bottom.

KeepassX 1 database actually lets the user sort the items in a custom order. And it is possible to sort it alphabetically. The latter is quite easy to implement. So I decided to look into it and will add it soon.

**Comment by tangrit** [comment]
Hi Marko,

thank you for implementation of sort.
I tried to compile the program for myself (my really first sailfish compilation) and I got the rpm-package after some warnings. After installation I have version 1.1.5 at my jolla, it starts, finds the last opened db, I can read the help text but the program stays in an endless loop diplaying "Decrypting Keepass database".

Trying to open a kdbx-file has the same effect. May maybe any impact of the preparation for keepass2 ?
